proxy/src/telemetry/metrics/labels.rs

```diff
@@ -1,12 +1,49 @@
-use std::collections::HashMap;
-use std::fmt::{self, Write};
-use std::hash;
-use std::sync::Arc;
+use std::{
+    collections::HashMap,
+    fmt::{self, Write},
+    hash,
+    path::PathBuf,
+    sync::Arc,
+};
 
 use http;
 
 use ctx;
 use telemetry::event;
+use transport::tls;
+
+macro_rules! mk_err_enum {
+    { $(#[$m:meta])* enum $name:ident from $from_ty:ty {
+         $( $from:pat => $reason:ident ),+
+     } } => {
+        $(#[$m])*
+        pub enum $name {
+            $( $reason ),+
+        }
+
+        impl fmt::Display for $name {
```

Comment:
Why not just use `libc::strerror_r`?

Reply:
IMO, this is less work than having to deal with C FFI for a fairly trivial `Display` implementation.